lib/puppet/provider/yumrepo/inifile.rb

```diff
@@ -74,8 +74,7 @@ def self.prefetch(resources)
   def self.reposdir(conf='/etc/yum.conf', dirs=['/etc/yum.repos.d', '/etc/yum/repos.d'])
     reposdir = find_conf_value('reposdir', conf)
     # Use directories in reposdir if they are set instead of default
-    dirs = reposdir.split(",").map(&:strip) if reposdir
-    
+    dirs = reposdir.split(/[,|\s]/).map(&:strip) if reposdir
```

Comment:
Good question, I think it does split and strip multiple whitespace characters:

```
>>> "foo \n  \r bar".split()
['foo', 'bar']
```

Also https://docs.python.org/3/library/stdtypes.html#str.split